core/lib/ReactCore_Internal.re

```diff
@@ -921,6 +933,303 @@ module Make = (Implementation: HostImplementation) => {
+      (
+        ~parentHostView,
+        ~instance: instance(state, action, elementType),
+        ~position,
+        ~f,
+      ) => {
+        let {id, component, subElements, instanceSubTree, _} = instance;
+        let newHostRoot =
+          switch (component.elementType) {
+          | React => parentHostView
+          | Host => f(parentHostView, id, subElements.make, position)
+          };
+        traverseRenderedElement(newHostRoot, instanceSubTree, f, 0);
+      };
+
+    let mountInstanceUtil =
+        (implementationFunc, parentHostView, id, make, position) =>
+      switch (getInstance(id)) {
+      | Some(_) => parentHostView
+      | None =>
+        let hostView = make();
+        memoizeInstance(id, hostView);
+        implementationFunc(
+          ~parent=parentHostView,
+          ~child=hostView,
+          ~position,
+        );
+        hostView;
+      };
+
+    let mountInstance = (~parentHostView, ~instance, ~position) =>
+      processInstance(
+        ~parentHostView,
+        ~instance,
+        ~position=0,
+        ~f=mountInstanceUtil(Implementation.mountChild),
+      );
+
+    let remountInstanceUtil =
+        (implementationFunc, parentHostView, id, make, position) =>
+      switch (getInstance(id)) {
+      | Some(hostView) =>
+        implementationFunc(
+          ~parent=parentHostView,
+          ~child=hostView,
+          ~position,
+        );
+        hostView;
+      | None => parentHostView
+      };
+
+    let remountInstance = (~parentHostView, ~instance, ~position as _) =>
+      processInstance(
+        ~parentHostView,
+        ~instance,
+        ~position=0,
+        ~f=remountInstanceUtil(Implementation.remountChild),
+      );
+
+    let unmountInstanceUtil = (parentHostView, id, _, _) =>
+      switch (getInstance(id)) {
+      | Some(hostView) =>
+        Implementation.unmountChild(~parent=parentHostView, ~child=hostView);
+        hostView;
+      | None =>
+        /* TODO Determine if/when this case is even possible? */
+        print_endline(
+          "Host view instance "
+          ++ string_of_int(id)
+          ++ " wasn't found during rendered element unmount",
+        );
+        parentHostView;
+      };
+
+    let unmountInstance = (~parentHostView, ~instance) =>
+      processInstance(
+        ~parentHostView,
+        ~instance,
+        ~position=0,
+        ~f=unmountInstanceUtil,
+      );
+
+    let rec applyUpdateLogEntry:
+      type state action elementType.
+        (
+          ~subtreeChange: option(UpdateLog.subtreeChange),
+          ~parentHostView: hostView,
+          ~oldInstance: instance(state, action, elementType),
+          ~newInstance: instance(state, action, elementType),
+          UpdateLog.subtreeChange
+        ) =>
+        unit =
+      (~subtreeChange, ~parentHostView, ~oldInstance, ~newInstance, entry) =>
+        switch (entry) {
+        /* The first `Nested update means our root hasn't changed, moving on to the subtree */
```

Comment:
I'm getting confused by the terminology. What is root?

Reply:
The root of the subtree change. (i.e, top-level/last element of the list)